common/src/test/org/apache/hive/common/util/TestDateParser.java

```diff
@@ -45,37 +45,29 @@ void checkInvalidCase(String strValue) {
 
   @Test
   public void testValidCases() throws Exception {
-    checkValidCase("1945-12-31", Date.valueOf("1945-12-31"));
-    checkValidCase("1946-01-01", Date.valueOf("1946-01-01"));
-    checkValidCase("2001-11-12", Date.valueOf("2001-11-12"));
-    checkValidCase("0004-05-06", Date.valueOf("0004-05-06"));
-    checkValidCase("1678-09-10", Date.valueOf("1678-09-10"));
-    checkValidCase("9999-10-11", Date.valueOf("9999-10-11"));
+    checkValidCase("1945-12-31", Date.of(1945,12,31));
+    checkValidCase("1946-01-01", Date.of(1946,1,1));
+    checkValidCase("2001-11-12", Date.of(2001,11,12));
+    checkValidCase("0004-05-06", Date.of(4,5,6));
+    checkValidCase("1678-09-10", Date.of(1678,9,10));
+    checkValidCase("9999-10-11", Date.of(9999,10,11));
 
     // Timestamp strings should parse ok
-    checkValidCase("2001-11-12 01:02:03", Date.valueOf("2001-11-12"));
+    checkValidCase("2001-11-12 01:02:03", Date.of(2001,11,12));
 
     // Leading spaces
-    checkValidCase(" 1946-01-01", Date.valueOf("1946-01-01"));
-    checkValidCase(" 2001-11-12 01:02:03", Date.valueOf("2001-11-12"));
+    checkValidCase(" 1946-01-01", Date.of(1946,01,01));
+    checkValidCase(" 2001-11-12 01:02:03", Date.of(2001,11,12));
   }
 
   @Test
   public void testParseDateFromTimestampWithCommonTimeDelimiter() {
-    for (String d : new String[] { "T", " ", "-", ".", "_", "" }) {
+    for (String d : new String[] { "T", " ", "-", ".", "_" }) {
```

Comment:
Yes, since it is absolutely misleading. Someone can say it is year 2023 August 3, other can say, it is year 301 August 23 and just drop the 20, as it is done now with the trailing characters, mentioned earlier. If we can consider dropping the trailing characters, why can not we consider doing the same with leading ones?
Please consider this example: "2023-08-21:02:03" .  Is it 2023 August 2 and 1:02:03, or 2023 August 21 and 02:03? Both interpretation can be argued, but a parsing should not rely on guessing, it must be on clear formatting.

Reply:
Since Hive Dates are expected to be in the default format (YYYY-MM-DD), I would also consider `2023-08-21:02:03` to mean August 21, of year 2023. This is not very ambiguous once we know the default format.

> If we can consider dropping the trailing characters, why can not we consider doing the same with leading ones?

HIVE-27586 was mainly introduced to retain backward compatibility with previous versions that were dropping trailing chars. Leading invalid chars were never accepted I think.